aten/src/ATen/native/nested/NestedTensorMath.cpp

```diff
@@ -1180,5 +1181,52 @@ Tensor _reshape_nested(const Tensor& self, IntArrayRef proposed_shape) {
   return output;
 }
 
+Tensor slice_nested(
+    const Tensor& self,
+    int64_t dim,
+    c10::optional<int64_t> start,
+    c10::optional<int64_t> end,
+    int64_t step) {
+  int64_t ndim = self.dim();
+  if (ndim == 0) {
+    TORCH_CHECK_INDEX(false, "slice() cannot be applied to a 0-dim tensor.");
+  }
+  dim = maybe_wrap_dim(dim, ndim);
+  TORCH_CHECK(dim > 0, "Nested tensor dimension 0 cannot be sliced");
```

Comment:
Is there a fundamental reason for that? Or just that this is not implemented here?

Reply:
There's no fundamental reason - in fact, I added an impl since it was easy to special case when dim=0.

This enables things like this:
```python
nt[:]
nt[:2]
nt[1:2]
```

Note that full advanced indexing still doesn't work since select for NT only supports dim 0. This still fails for now:
```python
nt[:, 1, 1]  # RuntimeError: NestedTensor can only be selected along dimension 0, got dimension 1 instead.
```